pkg/cedar/intake/biz.go

```diff
@@ -85,6 +48,71 @@ func translateBizCase(ctx context.Context, bc *models.BusinessCase) (*wire.Intak
 	}
 
 	// build the collection of embedded objects
+
+	// TODO: edit business case db model to hold solutions array instead of individual fields
+
+	// business solutions
+	// preferred (required)
+	preferredSolution := &wire.EASIBusinessSolution{
+		SolutionType:            pStr("preferred"),
+		Title:                   pStr(bc.PreferredTitle.ValueOrZero()),
+		Summary:                 pStr(bc.PreferredSummary.ValueOrZero()),
+		AcquisitionApproach:     pStr(bc.PreferredAcquisitionApproach.ValueOrZero()),
+		SecurityIsApproved:      pBool(bc.PreferredSecurityIsApproved),
+		SecurityIsBeingReviewed: pStr(bc.PreferredSecurityIsBeingReviewed.ValueOrZero()),
+		HostingType:             pStr(bc.PreferredHostingType.ValueOrZero()),
+		HostingLocation:         pStr(bc.PreferredHostingLocation.ValueOrZero()),
+		HostingCloudServiceType: pStr(bc.PreferredHostingCloudServiceType.ValueOrZero()),
+		HasUI:                   pStr(bc.PreferredHasUI.ValueOrZero()),
+		Pros:                    pStr(bc.PreferredPros.ValueOrZero()),
+		Cons:                    pStr(bc.PreferredCons.ValueOrZero()),
+		CostSavings:             pStr(bc.PreferredCostSavings.ValueOrZero()),
+	}
+
+	obj.BusinessSolutions = append(obj.BusinessSolutions, preferredSolution)
+
+	// TODO: do we need to check if alternative a and b are filled out?
```

Comment:
I think Alternative A is required, alternative B is not. However, if Alternative B is partially filled, it should be _completely_ filled. 

As for if we need to do this validation... maybe? We already validate the business case when it gets submitted (see appvalidation package) so it may be unnecessary to revalidate. Or it may be easy enough to write that validation that it's worth redoing.

Reply:
Alternative A used to be required but now it is not. We made that change during the six week governance work period. The partially filled out scenario was kind of what I was getting at when I talked about checking fields individually.

We didn't check them individually before but im also not opposed to doing it (same goes for re validating here as well as in app)